lib/src/chart/line_chart/line_chart_data.dart

```diff
@@ -236,6 +238,8 @@ class LineChartBarData {
   /// show indicators based on provided indexes
   final List<int> showingIndicators;
 
+  final CircularIntervalList<double> dashArray;
```

Comment:
external objects should not be exposed to the user directly,
because in some scenarios we may change or remove this object,
1. maybe in the future we have our drawing effects, and we decide to not using `path_drawing` anymore.
2. maybe `path_drawing` decides to change this object, then our users are exposed to that decision.
and other scenarios.
please decide a descriptive name for that object, and let's have our properties, then in the lower layer (drawing section) we will map it to the `CircularIntervalList` and make our dashed path.
and I have an idea in my head to have a class to support other types of drawing, dashed path, dotted path, ...

Reply:
@imaNNeoFighT I this obj is just a circular array with a double generic. I can define the same thing in the models of this project. I assume the dependency will accept it? I'm not sure

Reply:
2ed8a8a

Reply:
there is nothing I can do about this. the type that the dependency requires is strongly typed and is not iterable which means there's nothing I can transform into it or from it. the only way around this problem is to remove the dependency and bring it inline and modify it or otherwise somehow override the class it uses. it's too much work for such a small change

Reply:
ok I was losing my mind but I got it. I had to cast the inner object which was just a plain List, then invoke the constructor on that casted List. Dart is soooooo strongly typed. I am used to TypeScript

Reply:
here we go fd0b7f3